### PR TITLE
fix: Regression in temps introduced by gopsutil update

### LINKED ISSUE #11294: Undocumented regression in `temp` plugin
### Relevant telegraf.conf

```toml
[[inputs.temp]]
```

### Logs from Telegraf

```text
n/a
```

### System info

Telegraf 1.22.4 on Debian 11

### Docker

_No response_

### Steps to reproduce

1. Configure Telegraf with a bare `temp` input on a Linux system that has `hwmon` devices.

### Expected behavior

In older versions (at least Telegraf <=1.17.3 but I haven't exhaustively tested), each temperature from `/sys/class/hwmon/*/temp_*` is exposed as a stream with a `sensor` tag derived from the filename in `/sys`:

```
sensor=coretemp_core0_critalarm
sensor=coretemp_core0_input
sensor=coretemp_core0_max
sensor=coretemp_core1_crit
sensor=coretemp_core1_critalarm
sensor=coretemp_core1_input
sensor=coretemp_core1_max
...
```

This is also what is documented in the `temp` plugin's README.

### Actual behavior

In new builds (at least Telegraf >=1.22.4, sorry I have't bisected) only the `/sys/class/hwmon/*/temp_*_input` temperatures are exposed:

```
sensor=coretemp_core0
sensor=coretemp_core1
...
```

which means that the max/crit/etc. values are no longer being reported anywhere, nor are any sensors that don't follow the pattern of `temp_*_input`. It looks like this was broken when you pulled in https://github.com/shirou/gopsutil/pull/905

I'm not sure what you should do to fix this now, but metrics shouldn't have undocumented regressions in their structure like this between versions. Ideally you would revert back to the old behavior because there is now missing data, in addition to the schema change. It looks like `gopsutil` now exposes the max and crit levels as fields that you could turn into metrics, but the other values are just dropped by `gopsutil`.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by srebhan**
- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #11294 

This PR brings back critical (`_crit`) and high (`_max`) temperature levels lost during a gopsutil update. We cannot restore the critical-alarm threshold (`_critalarm`) as this information is not returned anymore.

Furthermore, this PR also introduces a `output_scheme` flag to allow to get the critical, high and actual temperatures as _fields_ in one metric per sensor instead of one metric per sensor and type.

### CODE REVIEW

File: plugins/inputs/temp/README.md [hunk 1]
```diff
@@ -32,9 +32,17 @@ wmic /namespace:\\root\wmi PATH MSAcpi_ThermalZoneTemperature
 
 ## Example Output
 
+For `output = "measurement" (default) the output will look like this
```
Comment (Hipska) [inline plugins/inputs/temp/README.md:35]:
```suggestion
For `output = "measurement"` (default) the output will look like this
```

File: plugins/inputs/temp/README.md [hunk 2]
```diff
@@ -32,9 +32,17 @@ wmic /namespace:\\root\wmi PATH MSAcpi_ThermalZoneTemperature
...
+
 ```shell
```
Comment (Hipska) [inline plugins/inputs/temp/README.md:37]:
```suggestion
```text
```

File: plugins/inputs/temp/README.md [hunk 3]
```diff
@@ -32,9 +32,17 @@ wmic /namespace:\\root\wmi PATH MSAcpi_ThermalZoneTemperature
...
 temp,sensor=coretemp_physicalid0_crit temp=100 1531298763000000000
 temp,sensor=coretemp_physicalid0_critalarm temp=0 1531298763000000000
 temp,sensor=coretemp_physicalid0_input temp=100 1531298763000000000
 temp,sensor=coretemp_physicalid0_max temp=100 1531298763000000000
 ```
+
+For `output = "field" the output will look like this
```
Comment (Hipska) [inline plugins/inputs/temp/README.md:44]:
```suggestion
For `output = "field"` the output will look like this
```

File: plugins/inputs/temp/README.md [hunk 4]
```diff
@@ -32,9 +32,17 @@ wmic /namespace:\\root\wmi PATH MSAcpi_ThermalZoneTemperature
...
+
+```shell
```
Comment (Hipska) [inline plugins/inputs/temp/README.md:46]: This is not a shell script
```suggestion
```text
```

File: plugins/inputs/temp/sample.conf [hunk 5]
```diff
@@ -1,3 +1,9 @@
 # Read metrics about temperature
 [[inputs.temp]]
-  # no configuration
+  ## Sets the output scheme.
+  ## Available values are
+  ##    measurement -- output one metric per type "critical", "high" and "temp"
+  ##                   with a concatenated sensor tag (default)
+  ##    field       -- output only one metric with "critical", "high" and "temp"
+  ##                   as fields
+  # output_scheme = "measurement"
```
Comment (Hipska) [inline plugins/inputs/temp/sample.conf:9]: Maybe add a newline?

**Review by Hipska — CHANGES_REQUESTED** [review]

File: plugins/inputs/temp/temp_nolinux.go [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+	"github.com/influxdata/telegraf"
+)
+
+func (t *Temperature) Gather(acc telegraf.Accumulator) error {
+	temps, err := t.ps.Temperature()
+	if err != nil {
+		if strings.Contains(err.Error(), "not implemented yet") {
+			return fmt.Errorf("plugin is not supported on this platform: %v", err)
+		}
+		return fmt.Errorf("error getting temperatures info: %s", err)
```
Comment (Hipska) [inline plugins/inputs/temp/temp_nolinux.go:19]:
```suggestion
		if strings.Contains(err.Error(), "not implemented yet") {
			return fmt.Errorf("plugin is not supported on this platform: %w", err)
		}
		return fmt.Errorf("error getting temperatures info: %w", err)
```

**Comment by powersj** [comment]
@srebhan given the user reported the fields are still missing in #11294, and you filed the upstream issue, should this be closed for now?

**Comment by srebhan** [comment]
Yeah seems like a good idea to close this PR. However, the upstream issue is sitting there for some time already...

**Pull request closed without merging**